url708d477f92_zSakare_SENG3011_tgz-pJ8-test_runner_TestRunnerJ8/src/main/parser/SircaCSVParser.java

```diff
@@ -0,0 +1,72 @@
+package main.parser;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import main.implementations.OrderBuilderImpl;
+import main.implementations.OrderbookImpl;
+import main.implementations.order.Order;
+import main.interfaces.OrderBuilder;
+import main.interfaces.Orderbook;
+import au.com.bytecode.opencsv.CSVReader;
+import au.com.bytecode.opencsv.bean.ColumnPositionMappingStrategy;
+import au.com.bytecode.opencsv.bean.CsvToBean;
+
+/**
+ * CSV parser class for reading in Sirca formatted orderbook data.
+ */
+public class SircaCSVParser {
+	
+	public static Orderbook input(String fileName) {
+		ColumnPositionMappingStrategy strat = new ColumnPositionMappingStrategy();
+		strat.setType(SircaOrder.class);
+		String[] columns = new String[] {"instrument", "date", "time", "recordType", "price", "volume", "undisclosedVolume", "value","qualifiers", "transactionId", "bidId","askId", "bidOrAsk","entryTime","oldPrice", "oldVolume", "buyerBrokerId", "sellerBrokerId" }; // the fields to bind do in your JavaBean
+		strat.setColumnMapping(columns);
+		 
+		CsvToBean csv = new CsvToBean();
+		 
+		String csvFilename = "./resources/" + fileName;
+		CSVReader csvReader = null;
+		try {
+			csvReader = new CSVReader(new FileReader(csvFilename), ',', '\'', 2);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *csvReader*. Execution paths that do not contain closure statements were detected. Ensure *csvReader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).